src/observables/observable_udp_socket.c

```diff
@@ -30,29 +30,24 @@ static void on_receive(uv_udp_t *handle, ssize_t nread, const uv_buf_t *buf, con
                        unsigned flags) {
     Observable *observable = (Observable *)handle->data;
 
-    if (addr == NULL) {
-        // FIXME: do not know why libuv does this. addr == NULL, nread == 0.
-        // We do not need to react on this.
+    if (nread == 0 && addr != NULL) { // an empty frame received
```

Comment:
Wont this will be cleaner to use inside nread ==0 branch for both "addr == and != NULL"?
Also please try to start comment from capital letter.